packages/timeline-state-resolver/src/integrations/tricaster/__tests__/triCasterInfoParser.spec.ts

```diff
@@ -0,0 +1,536 @@
+import { TriCasterInfoParser } from '../triCasterInfoParser'
+
+describe('TriCasterInfoParser', () => {
+	test('parseSwitcher returns info', () => {
+		const parser = new TriCasterInfoParser()
+
+		const info = parser.parseSwitcherUpdate(MOCK_SWITCHER_UPDATE)
+
+		expect(info).toEqual({
+			inputCount: 30,
+			dskCount: 3,
+			meCount: 6,
+			ddrCount: 3,
+		})
+	})
+
+	test('parseProductInformation returns info', () => {
+		const parser = new TriCasterInfoParser()
+
+		const info = parser.parseProductInformation(MOCK_PRODUCT_INFORMATION)
+
+		expect(info).toEqual({
+			productModel: 'TC2ELITE',
+			sessionName: 'TEST SESSION',
+			outputCount: 8,
+		})
+	})
+})
+
+const MOCK_SWITCHER_UPDATE = `<switcher_update main_source="INPUT2" preview_source="INPUT9" effect="C:\\ProgramData\\NewTek\\Effects\\Transitions\\Fades\\Non Additive Fade.trans" iso_label="INPUT 9" button_label="9">
```

Comment:
Yeah, I see what you mean that this is definately not ideal.
Off the top of my head: Maybe we could make a Builder that can build mock data for us. That would atleast makes it easier to make specific mock data of the switcher

Reply:
I would be more than happy to have a talk about it